lib/identity.js

```diff
@@ -51,14 +51,22 @@ bedrock.events.on('bedrock-mongodb.ready', function init(callback) {
  * @param callback(err, record) called once the operation completes.
  */
 api.insert = function(actor, identity, callback) {
+  var meta = {};
   async.auto({
     checkPermission: function(callback) {
       api.checkPermission(
         actor, PERMISSIONS.IDENTITY_INSERT, {resource: identity}, callback);
     },
-    insert: ['checkPermission', function(callback) {
+    emit: ['checkPermission', function(callback) {
+      bedrock.events.emit('bedrock-identity.insert', identity, meta, callback);
+    }],
+    insert: ['emit', function(callback) {
       identity = bedrock.util.clone(identity);
 
+      // remove plaintext sysPassword and sysPasscode
+      delete identity.sysPassword;
+      delete identity.sysPasscode;
```

Comment:
This is what I can say about that:  https://github.com/digitalbazaar/bedrock-idp/blob/master/lib/auth.js#L405

The old IdP identity code replaces the sysPass* in the identity with the hashed versions.  If these properties are are not deleted, plaintext versions of sysPass will be stored in the identity.